proxy/state.go

```diff
@@ -0,0 +1,9 @@
+package proxy
```

Comment:
yeh. Makes more sense in the handshake PR which some of these commits were pulled from. empty interfaces here but filled in there